testplan/web_ui/testing/src/AssertionPane/AssertionTypes/CopyButton.js

```diff
@@ -1,7 +1,77 @@
 import React, {Component} from 'react';
 import PropTypes from 'prop-types';
 import {Button} from 'reactstrap';
-import CopyToClipboard from 'react-copy-html-to-clipboard';
+import copy from 'copy-html-to-clipboard';
+
+class CopyToClipboard extends React.PureComponent {
+  static propTypes = {
+    text: PropTypes.any.isRequired,
+    children: PropTypes.element.isRequired,
+    onCopy: PropTypes.func,
+    options: PropTypes.shape({
+      debug: PropTypes.bool,
+      message: PropTypes.string,
+      asHtml: PropTypes.bool,
+      onlyHtml: PropTypes.bool,
+      canUsePrompt: PropTypes.bool,
+    })
+  };
+
+
+  static defaultProps = {
+    onCopy: undefined,
+    options: undefined
+  };
+
+
+  onClick = event => {
+    const {
+      text,
+      onCopy,
+      children,
+      options
+    } = this.props;
+
+    const elem = React.Children.only(children);
+
+    let textToCopy;
+    if (typeof text == 'function')
+        textToCopy = text();
+    else
+        textToCopy = text;
+
+    const result = copy(textToCopy, options);
+
+    if (onCopy) {
+      onCopy(textToCopy, result);
+    }
+
+    if (elem && elem.props) {
+      if (typeof elem.props.onClick === 'function') {
+        elem.props.onClick(event);
+      } else if (typeof elem.props.onTouchTap === 'function') {
+        elem.props.onTouchTap(event);
+      }
+    }
+    
+  };
+
+
+  render() {
+    const {
+      text: _text,
+      onCopy: _onCopy,
+      options: _options,
+      children,
+      ...props
+    } = this.props;
+    const elem = React.Children.only(children);
+
+    return React.cloneElement(
+      elem, {...props, onClick: this.onClick, onTouchTap: this.onClick}
+    );
+  }
+}
```

Comment:
It seems we will need to distribute the original project's license file as well, if we copy-paste their code.

Reply:
Thanks Yifan, very good point. The package is distributed under the MIT licence https://github.com/ukrbublik/react-copy-html-to-clipboard/blob/master/LICENSE. You are more experienced in this: what is the set of changes I need to make so that we are on the safe side?